packages/telemetry/src/flight-recorder.js

```diff
@@ -272,18 +272,18 @@ export const makeSimpleCircularBuffer = async ({
  * @param {Pick<Awaited<ReturnType<typeof makeSimpleCircularBuffer>>, 'writeCircBuf'>} circBuf
  */
 export const makeSlogSenderFromBuffer = ({ writeCircBuf }) => {
-  let writes = [];
+  let toWrite = Promise.resolve();
   const writeJSON = (obj, serialized = serializeSlogObj(obj)) => {
     // Prepend a newline so that the file can be more easily manipulated.
     const data = new TextEncoder().encode(`\n${serialized}`);
     // console.log('have obj', obj, data);
-    writes.push(writeCircBuf(data));
+    toWrite = toWrite.then(() => writeCircBuf(data));
```

Comment:
Just in case we start returning stuff from `writeCircBuf`, no need to hold onto it.

```suggestion
    toWrite = toWrite.then(() => { writeCircBuf(data); });
```

Reply:
`writeCircBuf` has to be awaited. So this would be,
```js
toWrite = toWrite.then(async () => {
  await writeCircBuf(data);
});
```
which is pretty clunky. The type hints are all `Promise<void>` so nothing should rely on a return value.

Reply:
Oh my bad! Yeah ok too clunky